src/libponyrt/gc/gc.c

```diff
@@ -10,6 +10,11 @@
 
 DEFINE_STACK(ponyint_gcstack, gcstack_t, void);
 
+static bool might_reference_actor(pony_type_t* t)
+{
+  return t != NULL && t->might_reference_actor;
```

Comment:
sometimes type is NULL because we are looking at an opaque object (for example `pony_trace` sends NULL for type).

Sending everything through this rather than playing whack-a-mole with finding anything where type might be NULL seems like a much better idea.